algorithms/src/snark/marlin/ahp/prover/round_functions/first.rs

```diff
@@ -74,13 +74,14 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
     #[allow(clippy::type_complexity)]
     pub fn prover_first_round<'a, R: RngCore>(
         mut state: prover::State<'a, F, MM>,
+        batch_sizes: impl Iterator<Item = (&'a CircuitId, &'a usize)>,
```

Comment:
Why are we passing this separately? Isn't it included in `state`?

Reply:
It is in state, but we created this iterator already earlier so I thought why not reuse it for line 138 below.

Reply:
Why are we passing this in if it's in state?